test/integration_test.py

```diff
@@ -735,34 +752,78 @@ def _assert_indices_exist(self, catalog: CatalogName):
         # Uses the indexer service account credentials, which should have access
         # to all sources.
         tdr_client = plugin.tdr
-        all_sources = {
+        configured_sources = {
             frozendict(sourceSpec=str(source_spec),
                        sourceId=tdr_client.lookup_source_id(source_spec))
             for source_spec in map(TDRSourceSpec.parse, config.tdr_sources(catalog))
         }
-        self.assertEqual(_list_sources(tdr_client._http_client), all_sources)
+        public_sources = self._list_sources(catalog)
+        with self._service_account_credentials():
+            self.assertEqual(self._list_sources(catalog), configured_sources)
+        with self._public_service_account_credentials():
+            self.assertEqual(self._list_sources(catalog), public_sources)
+        self.assertEqual(public_sources - configured_sources, set())
+
+        managed_access_source_ids = {
+            source['sourceId']
+            for source in configured_sources - public_sources
+        }
 
-        public_tdr_client = TDRClient.with_public_service_account_credentials()
-        unauthenticated_sources = _list_sources(http_client())
-        self.assertEqual(unauthenticated_sources,
-                         _list_sources(public_tdr_client._http_client))
-        self.assertTrue(unauthenticated_sources <= all_sources)
+        def _source_ids_from_hits(hits: JSONs) -> Set[str]:
+            return {one(bundle['sources'])['sourceId'] for bundle in hits}
+
+        hits = self._get_entities(catalog, 'bundles')
+        indexed_source_ids = _source_ids_from_hits(hits)
+        self.assertEqual(indexed_source_ids,
+                         {source['sourceId'] for source in public_sources})
+        source_filter = {'sourceId': {'is': list(managed_access_source_ids)}}
+        url = furl(config.service_endpoint(),
+                   path='index/bundles',
+                   args={
+                       'catalog': catalog,
+                       'filters': json.dumps(source_filter)
+                   }).url
+        response = self._get_url_json(url)
+        self.assertEqual(response['hits'], [])
+        self.assertEqual(response['pagination']['total'], 0)
+
+        with self._service_account_credentials():
+            response = self._get_url_json(url)
+        indexed_source_ids = _source_ids_from_hits(response['hits'])
+        self.assertEqual(indexed_source_ids, managed_access_source_ids)
+        managed_access_files = (
+            file['url']
+            for bundle in response['hits']
+            for file in bundle['files']
+        )
+        try:
+            file_url = first(managed_access_files)
+        except ValueError:
+            pass
+        else:
+            file_url = furl(file_url)
+            self.assertEqual(file_url.path.segments[:2], ['repository', 'files'])
+            file_url.path.segments.insert(0, 'fetch')
+            response = self._http.request('GET', str(file_url))
```

Comment:
Not what I had requested. Why can't we simply tell the client to not follow redirects? `_get_url` already has support for that.